code/game/machinery/suit_storage_unit.dm

```diff
@@ -332,6 +332,8 @@
 		I.play_tool_sound(user, I.tool_volume)
 
 /obj/machinery/suit_storage_unit/proc/store_item(obj/item/I, mob/user)
+	if(!user.canUnEquip(I))
```

Comment:
still not the proper way to handle it. `drop_item` can already function, safely, with `NO_DROP` things.